src/java.base/share/classes/java/util/zip/InflaterInputStream.java

```diff
@@ -205,6 +203,7 @@ public long skip(long n) throws IOException {
         ensureOpen();
         int max = (int)Math.min(n, Integer.MAX_VALUE);
         int total = 0;
+        byte[] b = new byte[512];
```

Comment:
n may be less than 512 so maybe the temporary array can be of length Math.min(max, 512).

Reply:
That's a good idea. I just pushed a update to this PR with this suggested change. Plus updated the copyright year too.

Reply:
I ran tier1, tier2 and tier3 tests with this change and they completed successfully. Alan, does the current state of the PR looks fine to you? Should I go ahead with the merge?